### PR TITLE
feat: drop support for multi-process mode

### PR DESCRIPTION
**Pull request by piotr-oles**
We drop multi-process mode to simplify code-base and focus more on
optimizing single-process mode.

BREAKING CHANGE: 🧨 Dropped support for multi-process mode (removed `workers` option from
plugin configuration, `workersNumber` argument from serviceStart hook
and pre-computed consts: ONE_CPU, ALL_CPUS, ONE_CPU_FREE, TWO_CPUS_FREE)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,20 +197,12 @@ new ForkTsCheckerWebpackPlugin({
...
-
 - **vue** `boolean | { enabled: boolean, compiler: string }`:
   If `true` or `enabled: true`, the linter and compiler will process VueJs single-file-component (.vue) files. See the
   [Vue section](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin#vue) further down for information on how to correctly setup your project.
 
 - **useTypescriptIncrementalApi** `boolean`:
-  If true, the plugin will use incremental compilation API introduced in TypeScript 2.7. In this mode you can only have 1
-  worker, but if the changes in your code are small (like you normally have when you work in 'watch' mode), the compilation
-  may be much faster, even compared to multi-threaded compilation. Defaults to `true` when working with TypeScript 3+ and `false` when below 3. The default can be overridden by directly specifying a value.
+  If true, the plugin will use incremental compilation API introduced in TypeScript 2.7. Defaults to `true` when working with TypeScript 3+ and `false` when below 3. The default can be overridden by directly specifying a value.
```
Comment (johnnyreilly) [inline README.md:205]: Could we introduce a comment here to indicate that `useTypescriptIncrementalApi` does not support Vue please? I thought we already had this but I was mistaken 😄

**Review by johnnyreilly — CHANGES_REQUESTED** [review]

**Comment by johnnyreilly** [comment]
Looks good!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ jobs:
         os: 
           - ubuntu-latest
           - macos-latest
-        #  - windows-latest # to be uncommented when we drop multi-process mode
+          - windows-latest
```
Comment (johnnyreilly) [inline .github/workflows/main.yml:45]: This makes me so happy 🥰

**Review by johnnyreilly — CHANGES_REQUESTED** [review]

**Comment by phryneas** [comment]
On first view, looks good to me. Great move to get rid of some of this stuff :+1:

File: README.md (hunk 1 above)
Comment (piotr-oles) [reply README.md:205]: Are you sure? I saw the code that handles `vue` in the `CompilerHost.ts` that is used only by the `ApiIncrementalChecker.ts`

Comment (phryneas) [reply README.md:205]: It worked kind-of but primarily not really.
This was what #231 was meant to tackle, but as I had not enough feedback (or a project to test it on) back then and no time to finish it right now, it's still open :/

Comment (phryneas) [reply README.md:205]: (see also #219, #220

**Review by johnnyreilly — APPROVED** [review]
Good job!

**Pull request merged**